src/include/migraphx/algorithm.hpp

```diff
@@ -95,20 +95,24 @@ inline size_t levenshtein_distance(const std::string& s1, const std::string& s2)
     const size_t l1 = s1.length();
     const size_t l2 = s2.length();
 
-    std::vector<std::vector<size_t>> d(l1 + 1, std::vector<size_t>(l2 + 1));
+    std::vector<size_t> d0(l2 + 1), d1(l2 + 1);
```

Comment:
Should one of them be `l1` here ?

Reply:
No, this is correct. It is a modified algorithm. These are two identical sized vectors pertaining to just one of those strings: `s2`. Then we iterate through them in an inner loop (which is covered inside the outer loop by an index counter `1 .. l1`). Thanks.